devtools/gradle/src/main/java/io/quarkus/gradle/ResolvedGradleArtifactDeps.java

```diff
@@ -90,8 +90,12 @@ private void traverseDependencies(Set<? extends DependencyResult> dependencies,
         dependencies.forEach(dependency -> {
             if (dependency instanceof ResolvedDependencyResult) {
                 ResolvedComponentResult result = ((ResolvedDependencyResult) dependency).getSelected();
-                appDependencies.add(new AppDependency(toAppArtifact(result), "provided"));
-                traverseDependencies(result.getDependencies(), appDependencies);
+                //if the resolved component is platform-compile its a bom, which do not have a jar, lets ignore it
+                if (result.getVariants().size() <= 0
+                        || !result.getVariants().get(0).getDisplayName().equals("platform-compile")) {
+                    appDependencies.add(new AppDependency(toAppArtifact(result), "provided"));
```

Comment:
I'm not understanding why you check only the first Variant in the list?

And why is it added as `provided` ?

Reply:
Well, the honest answer is with my deep knowledge of Gradle I debugged the object layout and the resolved components only had one variant and for the bom it was named `platform-compile` and for the others it was `compile`. The bom should not be treated as a "normal" dependency afaik so that's why I omit it.
The `provided` keyword is added since that is what "works" inside Quarkus. The `implementation` configuration is the "closest" to `provided` (i guess `compileOnly` falls into that as well, but that won't work for tests). 
Imo the "problem" is that Quarkus is very hard-coded to Maven atm, we should try to change it so it's agnostic to whatever build-tool we use. - the only requirement should be using maven repos for dependencies.

Reply:
ok, so the `provided` thing isn't very convincing but we can go with that for now.

Regarding `getVariants().get(0)` : you might have multiple variants; it would be safer if you could loop on them all so to find the ones having `platform-compile`?